### PR TITLE
Fix operator config

### PR DESCRIPTION
**Pull request by rehanjaved237**

### CODE REVIEW

**Comment by stakater-user** [comment]
@rehanjaved237 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/bc522e85e0767fc24d716f91bfa4efa100503d03/checks) has Failed!

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-01537cca`

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/onsi/ginkgo v1.14.0
 	github.com/onsi/gomega v1.10.1
 	github.com/operator-framework/operator-sdk v0.18.1
+	github.com/prometheus/common v0.9.1
```
Comment (ahmedwaleedmalik) [inline go.mod:10]: Please run `go mod tidy` once

File: main.go [hunk 2]
```diff
@@ -32,17 +32,23 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/prometheus/common/log"
```
Comment (ahmedwaleedmalik) [inline main.go:35]: Don't use `log` from this module

File: main.go [hunk 3]
```diff
@@ -138,10 +138,16 @@ func getWatchNamespace() (string, error) {
...
 }
 
-func readSlackTokenSecret(config *config.Config, k8sReader client.Reader) string {
-	secretName := config.Slack.APIToken.SecretName
-	secretKey := config.Slack.APIToken.Key
+func getConfigSecretName() string {
+	configSecretName, _ := os.LookupEnv("CONFIG_SECRET_NAME")
+	if len(configSecretName) == 0 {
+		configSecretName = SlackDefaultSecretName
+		log.Info("CONFIG_SECRET_NAME is unset, using default value: " + SlackDefaultSecretName)
```
Comment (ahmedwaleedmalik) [inline main.go:145]: Replace `log` with `setupLog`

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Please move the methods `getConfigSecretName` and `readSlackTokenSecret` to `config/config.go` file. And remove any redundant methods from there

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-aa6c5b89`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-c8ca557c`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-1b8ac55d`

File: config/manager/kustomization.yaml [hunk 4]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: stakater/slack-operator
+  newTag: SNAPSHOT-PR-11-c8ca557c
```
Comment (rasheedamir) [inline config/manager/kustomization.yaml:8]: hard code tag name?

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-17efd9d6`

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-84ab8c35`

**Review by ahmedwaleedmalik — APPROVED** [review]
LGTM

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-49a21a26`

**Comment by stakater-user** [comment]
@ahmedwaleedmalik Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-11-f0f5e770`

**Pull request merged**